spmd/compiler/fusion.py

```diff
@@ -78,20 +103,44 @@ def update_info(self, gm: fx.GraphModule) -> None:
 
 
 def _insert_fusion_buffer_node(
-    gm: fx.GraphModule, insert_before_node: fx.Node, buffer_size: Iterable[int]
+    gm: fx.GraphModule,
+    buffer_size: int,
+    gi: Optional[GraphInfo] = None,
 ) -> fx.Node:
-    """insert a torch.empty node in front of insert_before_node"""
+    """insert a torch.empty node for the global buffer.
+    defaults to first node after placeholder nodes.
+    appends to GlobalInfo if passed in"""
+
+    # default to inserting just after last placeholder node
+    for node in gm.graph.nodes:
+        if node.op == OP.PLACEHOLDER:
+            continue
+        insert_before_node = node
+        break
+
+    # TODO - fix with correct rank - needs to match with higher DTensor device
+
+    rank = dist.get_rank()
+    if torch.distributed.is_initialized():
+        torch.cuda.set_device(rank)
+    rank_device = torch.cuda.current_device()
+
     with gm.graph.inserting_before(insert_before_node):
         new_buffer_node = gm.graph.create_node(
             OP.CALL_FUNCTION,
             target=torch.empty,
```

Comment:
did you consider the tradeoff between calling torch.empty every iteration, vs allocating the memory at compile time and stashing the reusable buffer as an attribute of the graphmodule?